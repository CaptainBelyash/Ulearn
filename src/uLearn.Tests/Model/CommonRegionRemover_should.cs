using System.Collections.Generic;
using System.IO;
using System.Linq;
using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using uLearn.Model.Blocks;

namespace uLearn.Model
{
	[TestFixture]
	public class CommonRegionRemover_should
	{
		private static readonly DirectoryInfo dir = new DirectoryInfo("tests");

		private static string RemoveLabels(string file, params string[] labels)
		{
			var code = dir.GetFile(file).ContentAsUtf8();
			var remover = new CommonRegionRemover();
			IEnumerable<Label> notRemoved;
			code = remover.Remove(code, labels.Select(s => new Label { Name = s }), out notRemoved);
			Assert.IsEmpty(notRemoved);
			return code;
		}

		[Test, UseReporter(typeof(DiffReporter))]
		public static void RemoveRegion()
		{
			Approvals.Verify(RemoveLabels("OverloadedMethods.cs", "methods_1_and_2"));
		}

		[Test, UseReporter(typeof(DiffReporter))]
		public static void RemoveEmbeddedRegions()
		{
			Approvals.Verify(RemoveLabels("regions.txt", "b", "c"));
		}

		[Test, UseReporter(typeof(DiffReporter))]
		public static void RemoveOverlapedRegions()
		{
			Approvals.Verify(RemoveLabels("regions.txt", "d", "e"));
		}

		[Test, UseReporter(typeof(DiffReporter))]
		public static void RemoveRegions()
		{
			Approvals.Verify(RemoveLabels("regions.txt", "f", "g", "h"));
		}

		[Test, UseReporter(typeof(DiffReporter))]
		public static void RemoveRegionsCode()
		{
			Approvals.Verify(RemoveLabels("OverloadedMethods.cs", "methods_1_and_2", "methods_2_and_3"));
		}
	}
}