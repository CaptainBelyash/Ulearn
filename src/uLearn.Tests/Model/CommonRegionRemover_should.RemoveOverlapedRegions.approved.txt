region b
region c
endregion c
endregion b


region f
region g
endregion f
region h
endregion g
endregion h